lib/zeitwerk/loader.rb

```diff
@@ -463,9 +463,10 @@ class << self
       #
       #   require "zeitwerk"
       #   loader = Zeitwerk::Loader.new
-      #   loader.tag = File.basename(__FILE__, ".rb")
-      #   loader.inflector = Zeitwerk::GemInflector.new
-      #   loader.push_dir(__dir__)
+      #   root_file = __FILE__
+      #   loader.tag = File.basename(root_file, ".rb")
+      #   loader.inflector = Zeitwerk::GemInflector.new(root_file)
+      #   loader.push_dir(File.dirname(root_file))
```

Comment:
@fxn  you wrote:
> Could you please use `__FILE__` in the constructor and keep `__dir__`?

sure. I wrote it that way because that's how it's used internally and seemed to me reasonable to align them in this way